File: LPAv/Trabalho 9 - Python/questao9.py
import re

stringDeEntrada = input()
while (stringDeEntrada != "####"):
    if (re.match("^RS[0-9]{1,3}(\\.[0-9]{3,3}){0,1},[0-9]{2,2}$", stringDeEntrada)):
        print("SIM")
    else:
        print("NAO")
    stringDeEntrada = input()

File: LPAv/Trabalho  9 - C++/questao9.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <regex>
using namespace std;

int main()
{
    string expressaoRegular;
    cin >> expressaoRegular;
    while (expressaoRegular != "####")
    {
        if (regex_match(expressaoRegular, regex("^RS[0-9]{1,3}(\\.[0-9]{3,3}){0,1},[0-9]{2,2}$")))
        {
            printf("SIM\n");
        }
        else
        {
            printf("NAO\n");
        }
        cin >> expressaoRegular;
    }
}